packages/devtools_app/lib/src/notification_provider.dart

```diff
@@ -0,0 +1,8 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// Interface for pushing notifications to the app.
+abstract class NotificationProvider {
```

Comment:
Can this interface be removed? I only see one implementation and no uses as an interface?

Reply:
Yes, it doesn't need to be in this PR.

However, I wanted to talk about what we can do with an interface for the notifications provider.

When we construct our controllers, we can give them a notifications interface (which has no Flutter dependencies like State) to use.  It will look like this:

```dart
file: controllers.dart

factory ProvidedControllers.defaults(BuildContext context) {
    return ProvidedControllers(
      inspector: InspectorController(
        notificationsProvider: Notifications.of(context),
        ...
      ),
      ...similarly...
    );
  }
```

This way, we can detect events that DevTools can help the user understand.

For example, when a widget overflows, the InspectorController can push a notification about the overflowed widget.  Then, Notifications can show the notification and include a callback to show the layout debugger tab.  If the user clicks the notification, they'll go right to where they can debug the app.